scripts/build-shotcut.sh

```diff
@@ -386,96 +385,81 @@ function set_globals {
-  REPOTYPES[4]="git"
-  REPOTYPES[5]="git"
-  REPOTYPES[6]="http-tgz"
-  REPOTYPES[7]="git"
-  REPOTYPES[8]="http-tgz"
-  REPOTYPES[9]="git"
-  REPOTYPES[10]="git"
-  REPOTYPES[11]="git"
-  REPOTYPES[12]="git"
-  REPOTYPES[13]="git"
-  REPOTYPES[14]="http-tgz"
-
-  # And, set up the revisions
-  REVISIONS[0]=""
+
+  #  Dependency revisions
+  #
+  #  In the block below locations and revisions for building shotcut can be
+  #  found. The format is (space-separated:)
+  #    - repository type, git/svn/http-tgz
+  #    - repository location, either a git remote or a url for a tarball
+  #    - SHA or tag (NOT a branch) from which to build this module
+  #    - FETCHFROM marker, pointing to the branch from which the SHA will be
+  #      taken from by the update-revs.sh script. If not included,
+  #      update-revs.sh will not touch this line.
+  #    - git describe for the current SHA. Not actually used for anything, but
+  #      including it here gives a nice diff showing both the old and new tag.
+  IDX=0
+  while read repotype repoloc reposha therestoftheline; do
+      REPOTYPES[$IDX]=$repotype
+      REPOLOCS[$IDX]=$repoloc
+      REVISIONS[$IDX]=$reposha
+      IDX=$(( $IDX + 1 ))
+  done <<EOF
+  git git://github.com/FFmpeg/FFmpeg.git origin/release/2.3
+  git git://github.com/mltframework/mlt.git 9359202c362bef3bedcd36d3d020d92586bca8f4 FETCHFROM:origin/master v0.9.6-16-g9359202
+  git git://github.com/ddennedy/frei0r.git baa08d27193d758b5139dfa503dbedc881298f1c FETCHFROM:origin/master v1.4-25-gbaa08d2
+  git git://repo.or.cz/x264.git 121396c71b4907ca82301d1a529795d98daab5f8 FETCHFROM:origin/master unknown-tag
+  git http://git.chromium.org/webm/libvpx.git 8ea7bafdaa5785bb916106a5a47105828c7836ad FETCHFROM:origin/master v1.3.0-5787-g8ea7baf
+  git http://git.sesse.net/movit/ b757b4f1ca57c6dcabfe4ad696a2b887544d1a71 FETCHFROM:origin/master 1.1.3
+  http-tgz http://downloads.sourceforge.net/project/lame/lame/3.99/lame-3.99.5.tar.gz lame-3.99.5
+  git git://github.com/mltframework/shotcut.git a1622c54fa72413bdb5ac115af1ef2b2b7ddba16 FETCHFROM:origin/master v15.04-9-ga1622c5
+  http-tgz http://ftp.us.debian.org/debian/pool/main/s/swh-plugins/swh-plugins_0.4.15+1.orig.tar.gz swh-plugins-0.4.15+1
+  git git://github.com/mltframework/webvfx.git a54b0938757d53033424f411a0292926249313ba FETCHFROM:origin/master 0.4.4-44-ga54b093
+  git git://github.com/georgmartius/vid.stab.git 4ec5be1ba92325de8154106a1470a49242459125 FETCHFROM:origin/master release-0.98a-14-g4ec5be1
+  git git://github.com/anholt/libepoxy.git 20062c25e7612cab023cdef44d3277ba1bd0b2de FETCHFROM:origin/master v1.2-13-g20062c2
+  git git://git.opus-codec.org/opus.git 36e0445e619a2814b05d4e48dbaee785b70aac4d FETCHFROM:origin/master draft-ietf-codec-oggopus-06-21-g36e0445
+  git https://github.com/videolan/x265 47b970f9f675addb74080c1073ef7035c8fd6e66 FETCHFROM:origin/master 1.6-126-g47b970f
+  http-tgz http://bitbucket.org/eigen/eigen/get/3.2.4.tar.gz eigen-eigen-10219c95fe65
+EOF
+
```

Comment:
This is messy and difficult to read. Why not just populate the "XXX_REVISION" fields above. It would be cleaner and easier to read. I guess it might require some regex magic in your script to update the revisions.